zio-http/src/main/scala/zio/http/api/TextCodec.scala

```diff
@@ -74,4 +80,20 @@ object TextCodec {
 
     def encode(value: UUID): String = value.toString
   }
+
+  case object EncodingCodec extends TextCodec[Encoding] {
```

Comment:
We do **not** want to add a new `TextCodec` for every structured type! There would be hundreds of `TextCodec` which would make maintenance impossible.

Rather, we want to use the *existing* text codecs, to make an `In`, and then `transform` that `In[String]` to `In[Encoding]` using the `In#transform` method. Does that make sense?

Reply:
it makes sense. I will update accordingly.

Reply:
updated